src/middleware/LoadAppealMiddleware.ts

```diff
@@ -29,34 +28,37 @@ export class LoadAppealMiddleware extends BaseMiddleware {
             const companyNumber = req.query[COMPANY_NUMBER_QUERY_KEY] as string;
             const appealId = req.query[APPEAL_ID_QUERY_KEY] as string;
 
-            const session = req.session.unsafeCoerce();
+            const session: Session | undefined = req.session;
 
-            const applicationData = session.getExtraData()
-                .chainNullable<ApplicationData>(extraData => extraData[APPLICATION_DATA_KEY])
-                .ifNothing(() => session.saveExtraData(APPLICATION_DATA_KEY, {}))
-                .orDefault({} as ApplicationData);
+            let applicationData: ApplicationData | undefined = session!.getExtraData(APPLICATION_DATA_KEY);
+
+            if (!applicationData) {
```

Comment:
Could be simplified like the other ones

Reply:
We dont need to set ExtraData if applicationData is defined